File: Structural/Bridge/Cpp/Radio.cpp
// Radio.cpp

#include "Radio.h"

void Radio::turnOn() {
  on = true;
  std::cout << "Radio is now ON." << std::endl;
}

void Radio::turnOff() {
  on = false;
  std::cout << "Radio is now OFF." << std::endl;
}

void Radio::setVolume(int volume) {
  this->volume = volume;
  std::cout << "Radio volume set to " << volume << std::endl;
}

int Radio::getVolume() const {
  return volume;
}

bool Radio::isOn() const {
  return on;
}

File: Structural/Bridge/CSharp/Radio.cs
// Radio.cs

using System;

public class Radio : IDevice {
  private bool isOn = false;
  private int volume = 30;

  public void TurnOn() {
    isOn = true;
    Console.WriteLine("Radio is now ON.");
  }

  public void TurnOff() {
    isOn = false;
    Console.WriteLine("Radio is now OFF.");
  }

  public void SetVolume(int volume) {
    this.volume = volume;
    Console.WriteLine($"Radio volume set to {volume}");
  }

  public int GetVolume() {
    return volume;
  }

  public bool IsOn() {
    return isOn;
  }
}
